mito.go

```diff
@@ -358,40 +370,44 @@ func compile(src, root string, fold bool, libs ...cel.EnvOption) (cel.Program, *
 		}
 	}
 
-	prg, err := env.Program(ast)
+	var progOpts []cel.ProgramOption
+	if details {
+		progOpts = []cel.ProgramOption{cel.EvalOptions(cel.OptTrackState)}
```

Comment:
What are the memory implications of turning this on? Like how big does the `EvalState` become if say the program is processing a large XML response?

Reply:
I would imagine this would be untenably large. The actual size would depend how many allocations were made to perform evaluations. In the worst case this would be the sum of all sizes for each evaluation node. In practice, I would imagine that this is mitigated by sharing between values when there has not been mutation or type conversion that requires a new allocation (e.g. `[]byte` → `string` or map construction, though in the latter, the only allocation would be the shallow value). I think we need to experiment to get a completish picture.

Reply:
> mitigated by sharing between values when there has not been mutation

That's what I hoping too, like a copy-on-write paradigm to achieve immutability while being memory efficient.